### PR TITLE
Remove references to GDAL & conversion service

### PR DESCRIPTION
**Pull request by tephenavies**
### What this PR does

* Removes references to conversion service in code and docs
* Adds an architecture decision record (ADR) to formalise the decision not to use the conversion service going forwards.

### Test me

Test the add data capabilities (local data and web data) at http://ci.terria.io/remove-convert/

### Checklist

-   [x] I've updated relevant documentation in `doc/`.
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've provided instructions in the PR description on how to test this PR.
- [x] Change the status of the ADR from Proposed to Accepted following review

### CODE REVIEW

File: lib/Models/Catalog/createCatalogItemFromFileOrUrl.ts
Comment (KeyboardSounds) [inline lib/Models/Catalog/createCatalogItemFromFileOrUrl.ts:34]: Not sure if this is a problem or not, but if I upload a random unsupported file (eg. a random PNG) I get "Failed to create reference" rather than a useful error message. I think this issue exists in `main` already though, so I won't block the PR.

**Review by KeyboardSounds — APPROVED** [review]

File: lib/Models/Catalog/createCatalogItemFromFileOrUrl.ts
Comment (KeyboardSounds) [reply lib/Models/Catalog/createCatalogItemFromFileOrUrl.ts:34]: Here I made an issue: https://github.com/TerriaJS/terriajs/issues/6197

Comment (tephenavies) [reply lib/Models/Catalog/createCatalogItemFromFileOrUrl.ts:34]: Yeah, the UrlReference errors aren't great at the moment.

I'm not sure if it was possible to get the conversion prompt/error in `main` (I tried a bit, but only got UrlReference errors), but after this PR you won't be able to get even more misleading errors about a conversion service.

**Comment by tephenavies** [comment]
For the sake of precautions (since I've merged from `main`) I'll let this build and then merge.

**Pull request merged**
